File: Longest-Substring-Without-Repeating-Characters/Longest-Substring-Without-Repeating-Characters.cpp

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans=0;
        int idx=-1;
        int location[256];
        memset(location,-1,sizeof(location));
        for(int i=0;i<s.length();i++)
        {
            if(location[s[i]]>idx)
                idx=location[s[i]];
            ans=max(ans,i-idx);
            location[s[i]]=i;
        }
        return ans;
    }
};


 

File: Longest-Substring-Without-Repeating-Characters/Longest-Substring-Without-Repeating-Characters.py


class Solution:
    # @return an integer
    def lengthOfLongestSubstring(self, s):
        ans = 0
        idx = -1
        location = [-1 for x in range(256)]
        for i in range(len(s)):
            if location[ord(s[i])]>idx:
                idx = location[ord(s[i])]
            if i-idx > ans:
                ans = i - idx
            location[ord(s[i])] = i
        return ans
 